### PR TITLE
Handle writing to a relative file

### PR DESCRIPTION
**Pull request by GregoryTravis**
### Pull Request Description

Fixes bug in writing to a non-absolute file (with backup).

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: test/Tests/src/System/File_Spec.enso [hunk 1]
```diff
@@ -670,6 +670,27 @@ spec =
             r2.should_fail_with File_Error
             r2.catch.should_be_a File_Error.Not_Found
 
+        Test.specify "Should handle parent-less file paths" <|
+            transient = enso_project.data / "transient"
+            f1 = transient / "./test1.txt"
+            f2 = File.new ("parentlesstest" + Date_Time.now.to_unix_epoch_milliseconds.to_text + ".txt")
```
Comment (radeusgd) [inline test/Tests/src/System/File_Spec.enso:676]: This is creating a new file in the Current Working Directory. Obviously we need this for the test as that's the specific scenario we check.

But I think we need to be more careful with the cleanup as we should try hard to not pollute the project root by accident. Currently, if the test fails, the cleanup will not run.

I suggest to use `Panic.with_finalizer` (essentially Enso's `finally` construct) to run the cleanup regardless of test success.

**Review by jdunkerley — APPROVED** [review]
Approving but agree with @radeusgd comment.

File: test/Tests/src/System/File_Spec.enso (hunk 1 above)
Comment (GregoryTravis) [reply test/Tests/src/System/File_Spec.enso:676]: Done

File: test/Tests/src/System/File_Spec.enso [hunk 2]
```diff
@@ -675,21 +675,23 @@ spec =
...
 
-            txt = "MY CONTENT"
-            txt.write f1
-            txt.write f2
+            cleanup =
+                f2.delete
 
-            f1.read_text . should_equal txt
-            f2.read_text . should_equal txt
+            Panic.with_finalizer cleanup <|
```
Comment (radeusgd) [inline test/Tests/src/System/File_Spec.enso]: IMHO for such simple case, I'd avoid the complexity of the 0-argument function block `cleanup` and just put the operation verbatim:
```suggestion
            Panic.with_finalizer f2.delete <|
```

But yours is OK too.

**Review by radeusgd — APPROVED** [review]

File: test/Tests/src/System/File_Spec.enso (hunk 2 above)
Comment (GregoryTravis) [reply test/Tests/src/System/File_Spec.enso]: Done

**Pull request merged**